Pool_Properties_Design.md

```diff
@@ -0,0 +1,213 @@
+# Design to Define Storage Pool Properties
+
+**Status**: Completed
+
+**Version**: Alpha
+
+**Author(s)**: Leon Wang (@leonwanghui)
+
+## Summary
+
+This proposal is designed for standardizing the properties definition of storage pool. This design is also part of referenced implementation of LineOfService defined in [Swordfish](https://www.snia.org/sites/default/files/SMI/swordfish/V105/Swordfish_v1.0.5_Specification.pdf) specification (v1.0.5), which promises to provide a unified approach for the management of storage and servers in hyperscale and cloud infrastructure environments.
+
+## Motivation
+
+As we all know, we did a lot of work to design OpenSDS Profile (see [here](https://docs.google.com/document/d/1irNnz019j0XuW6SZNigs6QuYFOCC3uL44EkRYJiHyq8/edit?usp=sharing)). In addition, the extensible [capabilties filter](https://github.com/opensds/opensds/pull/297) mechanism was also finished several days ago. This design spec proposes how to implement the StoragePool resources based on the Swordfish spec.
+
+Now that we have come to a conclusion about `Profile` design, it would be much easier to design `StoragePool` because both of them are composed of LineOfService defined in Swordfish.
+
+## Proposal
+
+### Goals
+
+* Redesign the definition of `ExtraSpec` contained in `StoragePoolSpec`.
+
+* Update all storage drivers to stay up with the changes.
+
+* Update the fake pool data declared in test package.
+
+### Data model impact
+
+This proposal will update the definition of `StoragePoolSpec` with these changes below:
+
+* Implement `DataStorageLoS`, `IOConnectivityLoS` and `DataProtectionLos` struct which defined in Swordfish:
+```go
+// DataStorageLoS can be used to describe a service option covering storage
+// provisioning and availability.
+type DataStorageLoS struct {
+	// The enumeration literal specifies the time after a disaster that the
+	// client shall regain conformant service level access to the primary
+	// store.
+	// The expectation is that the services required to implement this
+	// capability are part of the advertising system.
+	// +units:min
+	RecoveryTimeObjective int64 `json:"recoveryTimeObjective,omitempty" yaml:"recoveryTimeObjective,omitempty"`
+
+	// ProvisioningPolicy only supports "Fixed" and "Thin".
```

Comment:
Why "Fixed" here instead of "Thick" ?

Reply:
"Fixed" is used in the Swordfish spec.  It has the same meaning as "Thick".